### ISSUE TITLE
`step_population_scaling()` includes `cases_scaled` as a predictor in model even if min lag != 0

### ISSUE THREAD

**Issue by rachlobay** [cmu-delphi/epipredict#212]
The `step_population_scaling()` currently includes `cases_scaled` as a predictor in model even if min lag != 0. For example, if we look at the wf from the ex. in the function documentation (where we should note that only lag 7, 14 are specified for `cases_scaled` in the recipe):

```
library(epiprocess)
library(epipredict)
jhu <- epiprocess::jhu_csse_daily_subset %>%
  dplyr::filter(time_value > "2021-11-01", geo_value %in% c("ca", "ny")) %>%
  dplyr::select(geo_value, time_value, cases)

pop_data = data.frame(states = c("ca", "ny"), value = c(20000, 30000))

r <- epi_recipe(jhu) %>%
  step_population_scaling(df = pop_data,
                          df_pop_col = "value",
                          by = c("geo_value" = "states"),
                          cases, suffix = "_scaled") %>%
  step_epi_lag(cases_scaled, lag = c(7, 14)) %>% # here's where we specify lags
  step_epi_ahead(cases_scaled, ahead = 7, role = "outcome") %>%
  step_epi_naomit()

f <- frosting() %>%
  layer_predict() %>%
  layer_threshold(.pred) %>%
  layer_naomit(.pred) %>%
  layer_population_scaling(.pred, df = pop_data,
                           by =  c("geo_value" = "states"),
                           df_pop_col = "value")

wf <- epi_workflow(r, parsnip::linear_reg()) %>%
  fit(jhu) %>%
  add_frosting(f)

wf
```
The model output is 

```
Call:
stats::lm(formula = ..y ~ ., data = data)

Coefficients:
        (Intercept)         cases_scaled   lag_7_cases_scaled  lag_14_cases_scaled  
           -0.08908              2.62135             -0.38091             -0.59265  
```
When we strip the ex. of scaling, we can see that we don't get have a `cases` predictor in the model. That is, if we run

```
library(epiprocess)
library(epipredict)
jhu <- epiprocess::jhu_csse_daily_subset %>%
    dplyr::filter(time_value > "2021-11-01", geo_value %in% c("ca", "ny")) %>%
    dplyr::select(geo_value, time_value, cases)

pop_data = data.frame(states = c("ca", "ny"), value = c(20000, 30000))

r <- epi_recipe(jhu) %>%
    step_epi_lag(cases, lag = c(7, 14)) %>% # here's where we specify lags
    step_epi_ahead(cases, ahead = 7, role = "outcome") %>%
    step_epi_naomit()

f <- frosting() %>%
    layer_predict() %>%
    layer_threshold(.pred) %>%
    layer_naomit(.pred) 

wf <- epi_workflow(r, parsnip::linear_reg()) %>%
    fit(jhu) %>%
    add_frosting(f)

wf
```
The model output is
```
── Model ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Call:
stats::lm(formula = ..y ~ ., data = data)

Coefficients:
 (Intercept)   lag_7_cases  lag_14_cases  
  -7027.8481        2.4932        0.9628  
```
This issue is also present in the tests from `## Postprocessing` comment and beyond in the test-population_scaling.R file.

**Comment by rachlobay** [comment]
After looking over older issues, I think that Logan already identified this problem in #164. That said, I will leave this post up for another ex. of the problem.

**Comment by dajmcdon** [comment]
I believe this is fixed by #225
